spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/vision/CloudVisionAutoConfiguration.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.vision;
+
+import java.io.IOException;
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageAnnotatorSettings;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.DefaultCredentialsProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Provides Spring Beans for using Cloud Vision API.
+ *
+ * @author Daniel Zou
+ * @since 1.1
+ */
+@Configuration
+@EnableConfigurationProperties(CloudVisionProperties.class)
+@ConditionalOnClass(ImageAnnotatorClient.class)
+@ConditionalOnProperty(value = "spring.cloud.gcp.vision.enabled", matchIfMissing = true)
```

Comment:
@snicoll ,

Confirm, please, that we can't move such a property into the `@ConfigurationProperties` class.

See our discussion below in the `additional-spring-configuration-metadata.json` changes: https://github.com/spring-cloud/spring-cloud-gcp/pull/1194/files#r232061185

Thanks

Reply:
Yes. If you're not using the property in the API, it's basically dead code that shouldn't be there.

Reply:
OK. That is slightly different aspect though, but the question was if it really isn't going to work if we just place this property to the existing `@ConfigurationProperties` class as @dzou commented mentioning some open issue in Spring Boot: https://github.com/spring-projects/spring-boot/issues/2282

Aren't we going to have a generated `spring-configuration-metadata.json` in the META-INF based on that class anyway, which is going to be use in this `@ConditionalOnProperty`?

I'm just a bit confused that having `@ConfigurationProperties` with the prefix `spring.cloud.gcp.vision` we still need to add an entry to the `additional-spring-configuration-metadata.json` manually...

Reply:
I don't think that issue is related to what we're discussing here. It's about a custom condition that imports a `@ConfigurationProperties`. What we're discussing here is where the metadata should be generated. If you add an `enabled` flag in code but don't use it, the metadata will be generated automatically but you'd have effectively added dead code. I don't recommend you doing that.

I don't understand what's confusing. That property is not used in the code so it shouldn't be declared in the API. But you need to document it and that's what the additional metadata is for.

Makes sense?

Reply:
OK. Fair enough.

So, @dzou , feel free to merge.